### PR TITLE
Add config protos as described in Cfg doc

### PR DESCRIPTION
**Pull request by mandarjog**

### CODE REVIEW

File: istio/config/v1/aspect/listChecker/cfg.proto [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+syntax = "proto3";
+
+package istio.config.v1.aspect.listChecker;
+
+
+// Config - User visible configuration of this aspect
+// Must be defined as a proto
+message Config {
+  // checkBlackList determines if this behaves like a blacklist
+  bool checkBlackList=1;
```
Comment (geeknoid) [inline istio/config/v1/aspect/listChecker/cfg.proto]: Should have spaces around the = sign.

File: istio/config/v1/aspect/listChecker/cfg.proto [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  // List if defined inline
+  repeated string list=2;
```
Comment (geeknoid) [inline istio/config/v1/aspect/listChecker/cfg.proto]: The list and Provider URL fields don't seem like general features of the aspect, they are implementation-specific parameters.

File: istio/config/v1/aspect/listChecker/cfg.proto [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  // ProviderURL if list is externally maintained, url to it
+  string ProviderURL=3;
```
Comment (geeknoid) [inline istio/config/v1/aspect/listChecker/cfg.proto]: Should use lower case and underscores in protos.

File: istio/config/v1/cfg.proto [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+// Aspect is intent based. It specifies the intent "kind"
+message Aspect {
+  string kind = 1;
+  string adapter = 2; // optional, allows specifying an adapter
+  map<string, string> inputs = 3;
+  google.protobuf.Struct params = 4;
+}
+
+// cluster config defines adapters
```
Comment (geeknoid) [inline istio/config/v1/cfg.proto]: Wrong comment.

File: istio/config/v1/cfg.proto [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+message Adapter {
+	string name = 1;  // statsd-slow
+	string kind = 2;  // metrics
+	string impl = 3;  // istio.statsd
+	google.protobuf.Struct args = 4;    // based on impl {}
+}
+
+message ClusterConfig {
```
Comment (geeknoid) [inline istio/config/v1/cfg.proto]: Probably shouldn't have a k8s concept here (cluster)

File: istio/config/v1/cfg.proto [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+syntax = "proto3";
+
+import "google/protobuf/struct.proto";
+
+
+package istio.config.v1;
+
+// Configures a {single?} service
+message ServiceConfig {
```
Comment (douglas-reid) [inline istio/config/v1/cfg.proto]: we should add comments on all of the fields here, with as much documentation as we can.

Comment (douglas-reid) [inline istio/config/v1/cfg.proto]: Adding on. Having some inline comments with examples would be very helpful.

File: istio/config/v1/aspect/listChecker/cfg.proto (hunk 1 above)
Comment (mandarjog) [inline istio/config/v1/aspect/listChecker/cfg.proto]: done

File: istio/config/v1/aspect/listChecker/cfg.proto (hunk 2 above)
Comment (mandarjog) [inline istio/config/v1/aspect/listChecker/cfg.proto]: done. removed both.

File: istio/config/v1/cfg.proto (hunk 4 above)
Comment (mandarjog) [inline istio/config/v1/cfg.proto]: done

File: istio/config/v1/cfg.proto (hunk 5 above)
Comment (mandarjog) [inline istio/config/v1/cfg.proto]: changed to GlobalConfig

File: istio/config/v1/cfg.proto (hunk 6 above)
Comment (mandarjog) [inline istio/config/v1/cfg.proto]: Added comments and examples.

**Comment by mandarjog** [comment]
Updated. PTAL.

**Pull request merged**